main/inc/lib/AnnouncementManager.php

```diff
@@ -1913,7 +1913,7 @@ public static function getAnnouncements(
         while ($row = Database::fetch_array($result, 'ASSOC')) {
             if (!in_array($row['id'], $displayed)) {
                 $actionUrl = api_get_path(WEB_CODE_PATH).'announcements/announcements.php?'
-                    .api_get_cidreq_params($courseInfo['code'], $session_id, $row['to_group_id']);
+                    .api_get_cidreq_params($courseInfo['code'], $session_id);
```

Comment:
This should simply use $group_id, which is defined by the page context, not by the announcement properties.
If you are in the main announcement tool, $group_id = 0 so you don't switch context to the group.
If you are in the group 1's announcement section, then $group_id = 1 so when you look at the announcement, the interface still shows you're in the group, and when you click "back", you return to the group list.

Reply:
Understood, added $group_id to the go back action parameters & tested